src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts

```diff
@@ -70,53 +67,72 @@ export const useTradeRoutes = (
   const handleSellClick = useCallback(
     async (asset: Asset) => {
       try {
-        if (buyAsset.currency && asset.assetId === buyAsset.currency.assetId)
-          setValue('buyAsset.currency', sellAsset.currency)
-        const action = buyAsset.amount ? TradeActions.SELL : undefined
-        setValue('sellAsset.currency', asset)
-        setValue('buyAsset.amount', '')
-        setValue('action', action)
-        setValue('quote', undefined)
-        await getQuote({
-          amount: sellAsset.amount ?? '0',
-          sellAsset,
-          buyAsset,
+        const previousSellAsset = { ...getValues('sellAsset') }
+        const previousBuyAsset = { ...getValues('buyAsset') }
+
+        // Handle scenario where same asset is selected for buy and sell
+        if (asset.assetId === previousBuyAsset?.asset.assetId) {
+          setValue('sellAsset.asset', asset)
+          setValue('buyAsset.asset', previousSellAsset.asset)
+        } else {
+          setValue('sellAsset.asset', asset)
+          setValue('buyAsset.asset', buyTradeAsset?.asset)
+        }
+        updateQuote({
+          forceQuote: true,
+          amount: bnOrZero(sellTradeAsset?.amount).toString(),
+          sellAsset: asset,
+          buyAsset: buyTradeAsset?.asset,
           feeAsset,
-          action,
+          action: TradeAmountInputField.SELL,
         })
       } catch (e) {
         console.warn(e)
       } finally {
-        history.push('/trade/input')
+        history.push(TradeRoutePaths.Input)
       }
     },
-    [buyAsset, sellAsset, feeAsset, history, setValue, getQuote],
+    [
+      getValues,
+      updateQuote,
+      sellTradeAsset?.amount,
+      buyTradeAsset?.asset,
+      feeAsset,
+      setValue,
+      history,
+    ],
   )
 
   const handleBuyClick = useCallback(
     async (asset: Asset) => {
       try {
-        if (sellAsset.currency && asset.assetId === sellAsset.currency.assetId)
-          setValue('sellAsset.currency', buyAsset.currency)
-        const action = sellAsset.amount ? TradeActions.BUY : undefined
-        setValue('buyAsset.currency', asset)
-        setValue('sellAsset.amount', '')
-        setValue('action', action)
-        setValue('quote', undefined)
-        await getQuote({
-          amount: buyAsset.amount ?? '0',
-          sellAsset,
-          buyAsset,
+        const previousSellAsset = { ...getValues('sellAsset') }
+        const previousBuyAsset = { ...getValues('buyAsset') }
+
+        // Handle scenario where same asset is selected for buy and sell
+        if (asset.assetId === previousSellAsset?.asset.assetId) {
+          setValue('buyAsset.asset', asset)
+          setValue('sellAsset.asset', previousBuyAsset.asset)
+        } else {
+          setValue('buyAsset.asset', asset)
+          setValue('sellAsset.asset', sellTradeAsset?.asset)
+        }
```

Comment:
same deal, this is all redundant if we don't give the user the option to select the same asset